src/main/java/org/broadinstitute/hellbender/engine/TwoPassReadWalker.java

```diff
@@ -59,9 +63,11 @@ private void traverseReads(CountingReadFilter countedFilter, GATKApply f) {
     }
 
     /**
-     * Process an individual read (with optional contextual information). Must be implemented by tool authors.
-     * In general, tool authors should simply stream their output from apply(), and maintain as little internal state
-     * as possible. Indicates what actions are to be taken on the first iteration through the reads.
+     * Process an individual read (with optional contextual information) on the first pass through the reads. Must be
+     * implemented by tool authors.
+     *
+     * Since the {@link TwoPassReadWalker} is inherently stateful, any necessary state should be accumulated by this
+     * method during the first pass.
      *
      * @param read current read
      * @param referenceContext Reference bases spanning the current read. Will be an empty, but non-null, context object
```

Comment:
Add a link to {@link TwoPassReadsWalker#secondPassApply} for more description about what the referenceContext does. Or perhaps provide a link to the ReadWalker.apply() documentation so there isn't a burden of repeated documentation?